src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/DependencyAnalyzerRunner.cs

```diff
@@ -14,15 +14,20 @@ public class DependencyAnalyzerRunner : IDependencyAnalyzerRunner
     {
         private readonly IPackageRestorer _packageRestorer;
         private readonly IEnumerable<IDependencyAnalyzer> _packageAnalyzers;
-
         private readonly ILogger<DependencyAnalyzerRunner> _logger;
 
-        public DependencyAnalyzerRunner(IPackageRestorer packageRestorer,
+        public DependencyAnalyzerRunner(
+            IPackageRestorer packageRestorer,
             IEnumerable<IDependencyAnalyzer> packageAnalyzers,
             ILogger<DependencyAnalyzerRunner> logger)
         {
+            if (packageAnalyzers is null)
+            {
+                throw new ArgumentNullException(nameof(packageAnalyzers));
+            }
+
             _packageRestorer = packageRestorer ?? throw new ArgumentNullException(nameof(packageRestorer));
-            _packageAnalyzers = packageAnalyzers ?? throw new ArgumentNullException(nameof(packageAnalyzers));
+            _packageAnalyzers = packageAnalyzers.OrderyByPrecedence();
```

Comment:
Super nit-pick - `packageAnalyzers?.OrderyByPrecedence() ?? throw new ArgumentNullException(nameof(packageAnalyzers));` would be a smaller change here. Doesn't matter much, though.